### PR TITLE
Add CentralityProducer to pp reco. AOD event content extension. Add ZDC to CentralityBinProducer.

### PR DESCRIPTION
**Pull request by jmartinb**
Add CentralityProducer to pp reco. Corresponding AOD event content extension and also for ZDC. Add ZDC to CentralityBinProducer.

See slides including event size increase estimations: https://twiki.cern.ch/twiki/pub/CMS/HiCentrality2016/CentralityInppReco.pdf

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
backport #16054

File: RecoHI/Configuration/python/customise_PPwithHI.py [hunk 1]
```diff
@@ -137,18 +139,52 @@ def storeCaloTowersAOD(process):
...
- 
-     process=addHIIsolationProducer(process)
-     process=storeCaloTowersAOD(process)
- 
-     return process
- 
+
+    process=addHIIsolationProducer(process)
+    process=storeCaloTowersAOD(process)
+    process=customiseRecoCentrality(process)
```
Comment (cvuosalo) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: Why is `addRhoproducer` missing here? What are the intended differences, if any, between this PR and #16054?

Comment (dgulhan) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: The rho producer should also be in 80X it was merged to 81X:
https://github.com/cms-sw/cmssw/pull/15233

It was supposed to be backported to 80X but it wasn't based on the history here:
https://github.com/cms-sw/cmssw/commits/CMSSW_8_0_X/RecoHI/Configuration/python/customise_PPwithHI.py

How can we backport it now shall I make a new PR?

Comment (dgulhan) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: @mandrenguyen suggested me to merge this one and make a new PR. I just need to know whether any other changes for this PR or is it stable now?

Comment (mandrenguyen) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: @slava77 @cvuosalo Do you prefer that we update this PR to add the rho producer or make a separate PR?  Thanks for spotting that this was missing.

Comment (cvuosalo) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: @mandrenguyen: I think I would prefer a separate PR. @slava77: What do you think?

Comment (slava77) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: fine with me either way

Comment (mandrenguyen) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: @slava77 @cvuosalo :  Ok we'll open a second PR with the rho stuff.  You can sign this one when ready :-)

Comment (cvuosalo) [inline RecoHI/Configuration/python/customise_PPwithHI.py]: @mandrenguyen: I will approve this PR when the rho PR is submitted.

**Comment by cvuosalo** [comment]
+1

For #16120 59053ec131ee7f3ab1be3af3a6ae7daef93b3487

Modifying Heavy Ions customization to add centrality reconstruction. There should be no change in monitored quantities for pp workflows.

#16054 is the 81X version of this PR, and it has already been merged.

The code changes match those in #16054, and Jenkins tests show no problems. Additional tests performed for #16054 also show no problems.

**Comment by davidlange6** [comment]
+1
